soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Select.scala

```diff
@@ -3,38 +3,50 @@ package com.socrata.soql.ast
 import scala.util.parsing.input.{NoPosition, Position}
 import com.socrata.soql.environment._
 import Select._
-import com.socrata.NonEmptySeq
+import com.socrata.soql.{BinaryTree, Compound, PipeQuery}
 
 /**
   * A SubSelect represents (potentially chained) soql that is required to have an alias
   * (because subqueries need aliases)
   */
-case class SubSelect(selects: NonEmptySeq[Select], alias: String)
+case class SubSelect(selects: BinaryTree[Select], alias: String)
 
 /**
   * All joins must select from another table. A join may also join on sub-select. A join on a sub-select requires an
   * alias, but the alias is optional if the join is on a table-name only (e.g. "join 4x4").
   *
-  *   "join 4x4" => JoinSelect(TableName(4x4, None), None) { aliasOpt = None }
-  *   "join 4x4 as a" => JoinSelect(TableName(4x4, a), None) { aliasOpt = a }
+  *   "join 4x4" => JoinSelect(Left(TableName(4x4, None))) { aliasOpt = None }
+  *   "join 4x4 as a" => JoinSelect(Left(TableName(4x4, a))) { aliasOpt = a }
   *   "join (select id from 4x4) as a" =>
-  *     JoinSelect(TableName(4x4, None), Some(SubSelect(List(_select_id_), a))) { aliasOpt = a }
+  *     JoinSelect(Right(SubSelect(List(_select_id_), a))) { aliasOpt = a }
   *   "join (select c.id from 4x4 as c) as a" =>
-  *     JoinSelect(TableName(4x4, Some(c)), Some(SubSelect(List(_select_id_), a))) { aliasOpt = a }
+  *     JoinSelect(Right(SubSelect(List(_select_id_), a))) { aliasOpt = a }
   */
-case class JoinSelect(fromTable: TableName, subSelect: Option[SubSelect]) {
+case class JoinSelect(subSelect: Either[TableName, SubSelect]) {
```

Comment:
`case class(just an Either)` feels weird to me; I think I'd prefer breaking it into a two-case hierarchy, but I donlt feel super-strongly on the subject.

Reply:
JoinSelect is reduced to just an Either.  Keeping it just make it a little easier in transition.  Can refactor in the future.